src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,135 @@
+                    stake: stakeRegistry.getStakeForOperatorIdForQuorumAtBlockNumber(operatorId, quorumNumber, blockNumber)
+                });
+            }
+        }
+            
+        return operators;
+    }
+
+    /**
+     * @notice this is called by the AVS operator to get the relevant indices for the checkSignatures function
+     * if they are not running an indexer    
+     * @param registryCoordinator is the registry coordinator to fetch the AVS registry information from
+     * @param referenceBlockNumber is the block number to get the indices for
+     * @param quorumNumbers are the ids of the quorums to get the operator state for
+     * @param nonSignerOperatorIds are the ids of the nonsigning operators
+     * @return 1) the indices of the quorumBitmaps for each of the operators in the @param nonSignerOperatorIds array at the given blocknumber
+     *         2) the indices of the total stakes entries for the given quorums at the given blocknumber
+     *         3) the indices of the stakes of each of the nonsigners in each of the quorums they were a 
+     *            part of (for each nonsigner, an array of length the number of quorums they were a part of
+     *            that are also part of the provided quorumNumbers) at the given blocknumber
+     *         4) the indices of the quorum apks for each of the provided quorums at the given blocknumber
+     */
+    function getCheckSignaturesIndices(
+        IBLSRegistryCoordinatorWithIndices registryCoordinator,
+        uint32 referenceBlockNumber, 
+        bytes calldata quorumNumbers, 
+        bytes32[] calldata nonSignerOperatorIds
+    ) external view returns (CheckSignaturesIndices memory) {
+        IStakeRegistry stakeRegistry = registryCoordinator.stakeRegistry();
+        CheckSignaturesIndices memory checkSignaturesIndices;
+
+        checkSignaturesIndices.nonSignerQuorumBitmapIndices = registryCoordinator.getQuorumBitmapIndicesByOperatorIdsAtBlockNumber(referenceBlockNumber, nonSignerOperatorIds);
+        checkSignaturesIndices.totalStakeIndices = stakeRegistry.getTotalStakeIndicesByQuorumNumbersAtBlockNumber(referenceBlockNumber, quorumNumbers);
+        
+        checkSignaturesIndices.nonSignerStakeIndices = new uint32[][](nonSignerOperatorIds.length);
+        for (uint i = 0; i < nonSignerOperatorIds.length; i++) {
+            uint192 nonSignerQuorumBitmap = 
+                registryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex(
+                    nonSignerOperatorIds[i], 
+                    referenceBlockNumber, 
+                    checkSignaturesIndices.nonSignerQuorumBitmapIndices[i]
+                );
+            // the number of quorums the operator was a part of that are also part of the provided quorumNumbers
+            checkSignaturesIndices.nonSignerStakeIndices[i] = new uint32[](BitmapUtils.countNumOnes(nonSignerQuorumBitmap & BitmapUtils.bytesArrayToBitmap(quorumNumbers)));
```

Comment:
is there a reason for moving the `BitmapUtils.bytesArrayToBitmap(quorumNumbers)` operation inside the loop in this commit https://github.com/Layr-Labs/eigenlayer-contracts/commit/f1faa75d5e0ac150dbaa1e7a1dcd17298317f2c3 ?
seems both more inefficient (I realize that's not important here) and maybe a bit harder to parse? AFAICT it's the same calculation/output for every loop iteration.

Reply:
stack too deep :(